src/lambda/gc02_check_group_access_configuration/app.py

```diff
@@ -0,0 +1,389 @@
+    has_admin_policy = next((p for p in managed_policies if p.get("PolicyName", "") == "AdministratorAccess"), False) or next((p for p in inline_policies if policy_doc_gives_admin_access(p.get("PolicyDocument", "\{\}"))), False)
+    has_non_admin_policy = next((p for p in managed_policies if p.get("PolicyName", "") != "AdministratorAccess"), False) or next((p for p in inline_policies if not policy_doc_gives_admin_access(p.get("PolicyDocument", "\{\}"))), False)
+    
+    # Does the group have admin policies and non admin policies?
+    if has_admin_policy and has_non_admin_policy:
+        # yes, there is a mixture of admin and non-admin roles attached to the group. Return NON_COMPLIANT evaluation for group
+        return build_evaluation(
+            group_name,
+            "NON_COMPLIANT",
+            event,
+            resource_type=GROUP_RESOURCE_TYPE,
+            annotation=f"Group '{group_name}' has attached policies that contain both admin and non-admin roles."
+        )
+    
+    # Is the group an admin group?
+    if has_admin_policy:
+        # yes, fetch group members and check against admin_accounts
+        group_members = fetch_group_members(group_name, event)
+        only_has_admins = next((m for m in group_members if m.get("UserName", "") not in admin_accounts), True)
+        if not only_has_admins:
+            return build_evaluation(
+                group_name,
+                "NON_COMPLIANT",
+                event,
+                resource_type=GROUP_RESOURCE_TYPE,
+                annotation=f"Group '{group_name}' is an admin group that contains non-admin members."
+            )
+            
+    annotation = f"Group '{group_name}' has policies that only provides admin roles and only has admin members." if has_admin_policy else f"Group '{group_name}' has policies that only provides non-admin roles."
+        
+    return build_evaluation(
+        group_name,
+        "COMPLIANT",
+        event,
+        resource_type=GROUP_RESOURCE_TYPE,
+        annotation=annotation
+    )
+    
+def fetch_group_members(group_name, event):
+    members = []
+    marker = None
+    try:
+        while True:
+            response = AWS_IAM_CLIENT.get_group(GroupName=group_name, Marker=marker) if marker else AWS_IAM_CLIENT.get_group(GroupName=group_name)
+            members = members + response.get("Users", [])
+            marker = response.get("Marker")
+            if not marker:
+                break
+    except botocore.exceptions.ClientError as ex:
+        if "NoSuchEntity" in ex.response['Error']['Code']:
+            ex.response['Error']['Message'] = f"Unable to fetch information for group '{group_name}'. No such entity found."
+        else:
+            ex.response["Error"]["Message"] = "InternalError"
+            ex.response["Error"]["Code"] = "InternalError"
+        raise ex
+    
+    return members
+        
+def policy_doc_gives_admin_access(policy_doc: str):
```

Comment:
What id the policy has multiple statements? Or if the Action or Resources sections are lists? Maybe try using the `iam_client.simulate_custom_policy()` API to check the permissions? I used it in GR12, seems to be good for this type of thing.

Reply:
How would this work with admin access? You would have to simulate all actions against everything and according to the docs wildcards aren't supported for this api

Reply:
Changed this to look for the equivalent of admin access in multiple statements. Don't think I can use the simulation though. This is likely the best we can do that is practical and doesn't require constant code updates